File: cpp/chapter4/binarySearch.cpp
#include<iostream>
using namespace std;

int binarySearch(int a[9], int x){
    int n=9;
    int left = 0, right = n-1, mid;

    while ( left < right ) {
        mid = ( left + right ) / 2;
        
        if ( x > a[mid] ) {
            left = mid+1;
        }
        else 
        {
            right = mid;
        }
    }
        if ( a[left] == x ) 
            return left;
        else 
            return -1;

}

int main(){
int a[9]={1,3,4,4,6,17,79,81,90};
int x = 709;

if (binarySearch(a,x) != -1)
    cout<<"The item was found"<<endl;
else
    cout<<"the item was not found"<<endl;

return 0;
}

File: python/chapter4/binarySearch.py

def binarySearch(a,x):
    n = len(a)
    left=0
    right=n-1
    while left < right :
        mid = int((left+right)/2)
        if (x > a[mid]):
            left = mid + 1
        else:
            right = mid
    if a[left] == x:
        return left
    else:
        return -1

def main():
    a =[1,3,4,4,6,17,79,81,90]
    if binarySearch(a,709) != -1:
        print("item was found!")
    else:
        print("item was not found!")

main()